src/gui/tray/ActivityItemActions.qml

```diff
@@ -0,0 +1,134 @@
+import QtQml 2.15
+import QtQuick 2.15
+import QtQuick.Controls 2.3
+import QtQuick.Layouts 1.2
+import Style 1.0
+
+RowLayout {
+    id: root
+
+    spacing: 20
+
+    property string objectType: ""
+    property variant activityLinks: []
+    property bool displayActions: false
+
+    property color moreActionsButtonColor: "transparent"
+
+    property int maxActionButtons: 0
+
+    property Flickable flickable
+
+    signal triggerAction(int actionIndex)
+
+    function actionButtonIcon(actionIndex, color) {
+        const verb = String(root.activityLinks[actionIndex].verb);
+        const objectType = String(root.objectType);
+        if (verb === "WEB" && (objectType === "chat" || objectType === "call" || objectType === "room")) {
+            return "image://svgimage-custom-color/reply.svg" + "/" + color;
+        }
+
+        return "";
+    }
+
+    function actionButtonText(actionIndex) {
+        const verb = String(root.activityLinks[actionIndex].verb);
+        const objectType = String(root.objectType);
+        if (verb === "DELETE") {
+            return qsTr("Mark as read")
+        } else if (verb === "WEB" && objectType !== "room" && (objectType === "chat" || objectType !== "call")) {
+            return qsTr("Reply")
+        }
+
+        return root.activityLinks[actionIndex].label;
+    }
+
+    Repeater {
+        // a max of maxActionButtons can will get dispayed as separate buttons
+        model: root.activityLinks.length > root.maxActionButtons ? 1 : root.activityLinks.length
+
+        ActivityActionButton {
+            id: activityActionButton
+
+            readonly property int actionIndex: model.index
+            readonly property bool primary: model.index === 0 && root.activityLinks[actionIndex].verb !== "DELETE"
+
+            Layout.minimumWidth: primary ? 100 : 80
```

Comment:
I think we should establish what these values actually represent, maybe in Style.qml? It's difficult to tell what they stand for in isolation and if we have other components with this sizing we might as well make that explicit with a shared binding